packages/devtools_app/lib/src/service/editor/api_classes.dart

```diff
@@ -0,0 +1,432 @@
+}
+
+/// A base class for all known events that an editor can produce.
+///
+/// The set of subclasses is not guaranteed to match actual events from any
+/// given editor as the editor might not implement all functionality or may be a
+/// future version running against an older version of this code/DevTools.
+sealed class EditorEvent with Serializable {
+  EditorEventKind get event;
+}
+
+/// An event sent by an editor when a new device becomes available.
+class DeviceAddedEvent extends EditorEvent {
+  DeviceAddedEvent({required this.device});
+
+  DeviceAddedEvent.fromJson(Map<String, Object?> map)
+      : this(
+          device:
+              EditorDevice.fromJson(map[_Field.device] as Map<String, Object?>),
+        );
+
+  @override
+  EditorEventKind get event => EditorEventKind.deviceAdded;
+
+  final EditorDevice device;
+
+  @override
+  Map<String, Object?> toJson() => {
+        _Field.device: device,
+      };
+
+  @override
+  bool operator ==(Object other) =>
+      other is DeviceAddedEvent &&
+      other.runtimeType == runtimeType &&
+      other.device == device;
+
+  @override
+  int get hashCode => device.hashCode;
+}
+
+/// An event sent by an editor when an existing device is updated.
+///
+/// The ID in this event always matches an existing device (that is, the ID
+/// never changes, or it would be considered a removal/add).
+class DeviceChangedEvent extends EditorEvent {
+  DeviceChangedEvent({required this.device});
+
+  DeviceChangedEvent.fromJson(Map<String, Object?> map)
+      : this(
+          device:
+              EditorDevice.fromJson(map[_Field.device] as Map<String, Object?>),
+        );
+
+  @override
+  EditorEventKind get event => EditorEventKind.deviceChanged;
+
+  final EditorDevice device;
+
+  @override
+  Map<String, Object?> toJson() => {
+        _Field.device: device,
+      };
+
+  @override
+  bool operator ==(Object other) =>
+      other is DeviceChangedEvent &&
+      other.runtimeType == runtimeType &&
+      other.device == device;
+
+  @override
+  int get hashCode => device.hashCode;
+}
+
+/// An event sent by an editor when a device is no longer available.
+class DeviceRemovedEvent extends EditorEvent {
+  DeviceRemovedEvent({required this.deviceId});
+
+  DeviceRemovedEvent.fromJson(Map<String, Object?> map)
+      : this(
+          deviceId: map[_Field.deviceId] as String,
+        );
+
+  @override
+  EditorEventKind get event => EditorEventKind.deviceRemoved;
+
+  final String deviceId;
+
+  @override
+  Map<String, Object?> toJson() => {
+        _Field.deviceId: deviceId,
+      };
+
+  @override
+  bool operator ==(Object other) =>
+      other is DeviceRemovedEvent &&
+      other.runtimeType == runtimeType &&
+      other.deviceId == deviceId;
+
+  @override
+  int get hashCode => deviceId.hashCode;
+}
+
+/// An event sent by an editor when the current selected device was changed.
+/// This could be as a result of the client itself calling the `selectDevice`
```

Comment:
nit: new line after first sentence of dart doc

Reply:
Done in 8ec9692b9